Move conformal maps to library (#48)
jeremylt/LFAToolkit.jl: docs/src/examples/advection.md

## Advection operator

This is an example of an advection operator by using polynomial bases vs nonpolynomial bases
The choices of nonpolynomial bases available are:
mapping = sausage_transformation(9)
mapping = kosloff_talezer_transformation(0.98)
mapping = hale_trefethen_strip_transformation(1.4)

### Problem formulation

The advection problem is given by

```math
\dfrac{\partial u}{\partial t} + c \dfrac{\partial u}{\partial x} = 0
```
where ``u = u(x,t)`` and ``c`` is its associated phase speed.

The weak form is given by

```math
\int_\Omega v \left(\dfrac{\partial u}{\partial t} \right) dV - \int_\Omega c u \left( \dfrac{\partial v}{\partial x} \right) dV = 0, \forall v \in V
```

for an appropriate test space ``V \subseteq H^1 \left( \Omega \right)`` on the domain.
In this weak formulation, boundary terms have been omitted, as they are not present on the infinite grid for Local Fourier Analysis.

For understanding about nonpolynomial bases, see paper Hale and Trefethen (2008) New quadrature formulas from conformal maps. https://doi.org/10.1137/07068607X

### LFAToolkit code

> The advection operator is a classical test case to see dispersion spectrum inside LFAToolkit.

Reviewer: `````suggestion
The advection operator is a classical test case to see dispersion spectrum inside LFAToolkit.

````@eval
using Markdown
Markdown.parse("""
```julia
$(read("../../../examples/ex010_advection.jl", String))
```
""")
````
`````

## Advection operator

This is an example of an advection operator by using polynomial bases vs nonpolynomial bases
The choices of nonpolynomial bases available are:
mapping = sausagetransformation(9)
mapping = koslofftalezertransformation(0.98)
mapping = haletrefethenstriptransformation(1.4)

### Problem formulation

The advection problem is given by

```math

> \dfrac{\partial \zeta}{\partial t} + U \dfrac{\partial \zeta}{\partial x} = 0

Reviewer 2: Convention in other examples is that $u$ is the independent variable.
```suggestion
\dfrac{\partial u}{\partial t} + c \dfrac{\partial u}{\partial x} = 0
```

```
where ``\zeta = \zeta(x,t)`` and ``U`` is its associated phase speed.

The weak form is given by

```math
\int_\Omega v \left(\dfrac{\partial \zeta}{\partial t} \right) dV - \int_\Omega U \zeta \left( \dfrac{\partial v}{\partial x} \right) dV = 0, \forall v \in V
```

> for an appropriate test space ``V \subseteq H_0^1 \left( \Omega \right)`` on the domain.

Reviewer 2: Usually $H_0^1$ would mean functions with zero trace on Dirichlet boundaries, but there are no such boundaries so $H^1$ is okay.
```suggestion
for an appropriate test space ``V \subseteq H^1 \left( \Omega \right)`` on the domain.
```

Technically, this is an abuse of notation because even $\cos x$ is not in $L^2([-\infty, \infty])$. This abuse is benign here.

In this weak formulation, boundary terms have been omitted, as they are not present on the infinite grid for Local Fourier Analysis.

For understanding about nonpolynomial bases, see paper New quadrature formulas from conformal maps

> SIAM J. NUMER. ANAL. Vol. 46, No. 2 (2008) by following Hale and Trefethen (2008)

Reviewer 2: @jeremylt I take it juliadoc doesn't support bibtex. Anyway, I think linking the DOI is most useful in this context.
```suggestion
For understanding about nonpolynomial bases, see paper Hale and Trefethen (2008) New quadrature formulas from conformal maps. https://doi.org/10.1137/07068607X
```

### LFAToolkit code

> The advection operator is a classical test case to see dipersion spectrum inside LFAToolkit.

Reviewer 2: ```suggestion
The advection operator is a classical test case to see dispersion spectrum inside LFAToolkit.
```
There are spell checking plugins for vscode, which you may want to activate.

## Advection operator

> This is an example of an advection operator by using polynomial bases vs nonpolynomial bases
> The choices of nonpolynomial bases available are:
> mapping = sausage_transformation(9)
> mapping = kosloff_talezer_transformation(0.98)
> mapping = hale_trefethen_strip_transformation(1.4)

Reviewer: ```suggestion
This is an example of the 1D scalar advection problem.
```

Reviewer: woops, typo in the suggestion - should be good now

### Problem formulation

The advection problem is given by

```math
\dfrac{\partial u}{\partial t} + c \dfrac{\partial u}{\partial x} = 0
```
where ``u = u(x,t)`` and ``c`` is its associated phase speed.

The weak form is given by

```math
\int_\Omega v \left(\dfrac{\partial u}{\partial t} \right) dV - \int_\Omega c u \left( \dfrac{\partial v}{\partial x} \right) dV = 0, \forall v \in V
```

for an appropriate test space ``V \subseteq H^1 \left( \Omega \right)`` on the domain.
In this weak formulation, boundary terms have been omitted, as they are not present on the infinite grid for Local Fourier Analysis.

> For understanding about nonpolynomial bases, see paper Hale and Trefethen (2008) New quadrature formulas from conformal maps. https://doi.org/10.1137/07068607X
> 

Reviewer: ```suggestion
```

### LFAToolkit code

> The advection operator is a classical test case to see dispersion spectrum inside LFAToolkit.

Reviewer: ```suggestion
The advection operator is a classical test case to see dispersion spectrum inside LFAToolkit.
Here we show the advection operator on a non-polynomial basis derived from the Hale-Trefethen strip transformation applied to a H1 Lagrange basis.

For understanding about nonpolynomial bases, see paper Hale and Trefethen (2008) New quadrature formulas from conformal maps. https://doi.org/10.1137/07068607X
```
